rdt/transformers/categorical.py

```diff
@@ -425,8 +437,33 @@ class LabelEncoder(BaseTransformer):
     values_to_categories = None
     categories_to_values = None
 
-    def __init__(self, add_noise=False):
+    def __init__(self, add_noise=False, order_by=None):
         self.add_noise = add_noise
+        if order_by not in [None, 'alphabetical', 'numerical_value']:
+            raise Error(
+                "order_by must be one of the following values: None, 'numerical_value' or "
+                "'alphabetical'"
+            )
+
+        self.order_by = order_by
+
+    def _order_categories(self, unique_data):
+        if self.order_by == 'alphabetical':
+            if unique_data.dtype.type not in [np.str_, np.object_]:
+                raise Error("The data must be of type string if order_by is 'alphabetical'.")
+
+        elif self.order_by == 'numerical_value':
+            if not np.issubdtype(unique_data.dtype.type, np.number):
+                raise Error(
+                    'The data must be numerical or able to be casted as a float if order_by '
+                    "is 'numerical_value'."
+                )
+
+        if self.order_by is not None:
+            unique_data = np.sort(unique_data[~pd.isna(unique_data)])
+            unique_data = np.append(unique_data, [np.nan])
```

Comment:
I think there is a misunderstanding of what this method does. It only takes in unique values as it is ordering the keys for the `self.values_to_categories` dict. The result of this method should be a list of all unique categories sorted, and with nan appearing last. That is why previously I was only adding it if there were actually nans, but truthfully there is no harm in always adding it

Reply:
Actually I will change it to only add if there is a nan because the spec is that any categories not seen in `fit` (including nans) should not be transformable